--- plank-0.11.89.orig/lib/PositionManager.vala
+++ plank-0.11.89/lib/PositionManager.vala
@@ -66,7 +66,7 @@
 			
 			// NOTE don't call update_monitor_geo to avoid a double-call of dockwindow.set_size on startup
 			var session=Environment.get_variable("XDG_CURRENT_DESKTOP");
-			if (session != null && session.contains("GNOME")) {
+			if (session != null && (session.contains("GNOME") || session.contains("Pantheon"))) {
 				screen.get_monitor_geometry (find_monitor_number (screen, controller.prefs.Monitor), out monitor_geo);
 			}
 			else {
@@ -141,7 +141,7 @@
 			var old_monitor_geo = monitor_geo;
 
 			var session=Environment.get_variable("XDG_CURRENT_DESKTOP");
-			if (session != null && session.contains("GNOME")) {
+			if (session != null && (session.contains("GNOME") || session.contains("Pantheon"))) {
 				screen.get_monitor_geometry (find_monitor_number (screen, controller.prefs.Monitor), out monitor_geo);
 			}
 			else {
